### PR TITLE
Nuttall window

### PR DESCRIPTION
**Pull request by alvgaona**
Relates #85366

### CODE REVIEW

**Comment by alvgaona** [comment]
@lezcano 😄

File: torch/signal/windows/windows.py [hunk 1]
```diff
@@ -759,3 +761,64 @@ def general_hamming(M,
                           layout=layout,
                           device=device,
                           requires_grad=requires_grad)
+
+
+@_add_docstr(
+    r"""
+Computes the minimum 4-term Blackman-Harris window according to Nuttall.
```
Comment (lezcano) [inline torch/signal/windows/windows.py:768]: Formula?

File: torch/signal/windows/windows.py [hunk 2]
```diff
@@ -759,3 +761,64 @@ def general_hamming(M,
...
+
+{normalization}
+
+Arguments:
+    {M}
+
+Keyword args:
+    alpha (float, optional): the window coefficient. Default: 0.54.
```
Comment (lezcano) [inline torch/signal/windows/windows.py:776]: This function does not take an alpha parameter?

File: torch/signal/windows/windows.py [hunk 3]
```diff
@@ -759,3 +761,64 @@ def general_hamming(M,
...
+    {sym}
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Returns:
+    {return}
```
Comment (lezcano) [inline torch/signal/windows/windows.py:784]: I was thinking when looking at the docs, do you think we should just remove the `Returns` field from all the windows? It adds quite a bit of vertical space (two new fields) and it provides little-to-none new info.

File: torch/testing/_internal/opinfo/definitions/signal.py [hunk 4]
```diff
@@ -824,4 +824,57 @@ def make_signal_windows_opinfo(
...
+        reference_inputs_func=reference_inputs_window,
+        error_inputs_func=error_inputs_window,
+        skips=(
+            DecorateInfo(
+                unittest.expectedFailure,
+                "TestDecomp",
+                "test_comprehensive",
+                dtypes=[torch.float16],
+                device_type="cpu",
+            ),
```
Comment (lezcano) [inline torch/testing/_internal/opinfo/definitions/signal.py:842]: I think we should disallow now that we can `float16` and `bfloat16` dtypes as they have all these weird edge cases. We can always enable them later again if people ask for them. What do you think?

**Review by lezcano — APPROVED** [review]
I left a few small comments here and there, but otherwise it looks good. Thank you for the continued effort pushing this module forward!

File: torch/signal/windows/windows.py (hunk 1 above)
Comment (alvgaona) [reply torch/signal/windows/windows.py:768]: Sure, it's a sum of cosines. I'll add it.

**Comment by alvgaona** [comment]
> I left a few small comments here and there, but otherwise it looks good. Thank you for the continued effort pushing this module forward!

Thanks. 😄

File: torch/signal/windows/windows.py [hunk 5]
```diff
@@ -759,3 +732,71 @@ def general_hamming(M,
...
                           device=device,
                           requires_grad=requires_grad)
+
+
+@_add_docstr(
+    r"""
+Computes the minimum 4-term Blackman-Harris window according to Nuttall.
+
+.. math::
+    w_n = 1 - 0.3635819 \cos{(z)} + 0.4891775 \cos{(2z)} - 0.1365995 \cos{(3z)} + 0.0106411 \cos{(4z)}
```
Comment (lezcano) [inline torch/signal/windows/windows.py:742]:
```suggestion
    w_n = \sum_{j=0}^{M - 1} 1 - 0.3635819 \cos{(z_j)} + 0.4891775 \cos{(2z_j)} - 0.1365995 \cos{(3z_j)} + 0.0106411 \cos{(4z_j)}
```

File: torch/signal/windows/windows.py [hunk 6]
```diff
@@ -759,3 +732,71 @@ def general_hamming(M,
...
+
+where
+
+.. math::
+    z = \frac{2 \pi j}{M}, j = 0 \cdots M - 1
```
Comment (lezcano) [inline torch/signal/windows/windows.py:747]:
```suggestion
where ``z_j = 2 π j / M``.
```
to save some vertical space.

File: torch/signal/windows/windows.py (hunk 5 above)
Comment (lezcano) [reply torch/signal/windows/windows.py:742]: nb. If it doesn't fit the formula in the width of the page, perhaps you can trim the last 1 or 2 digits from the decimal expansion of the coefficients?

**Review by lezcano — COMMENTED** [review]
Thank you for the clean-up! I think the formula has a couple errata though. Please have a look and tell me what you think.

File: torch/signal/windows/windows.py (hunk 5 above)
Comment (alvgaona) [reply torch/signal/windows/windows.py]: Trimmed 2 and now it fits. Let's see what the build renders.

File: torch/signal/windows/windows.py [hunk 7]
```diff
@@ -759,3 +732,67 @@ def general_hamming(M,
...
                           device=device,
                           requires_grad=requires_grad)
+
+
+@_add_docstr(
+    r"""
+Computes the minimum 4-term Blackman-Harris window according to Nuttall.
+
+.. math::
+    w_n = 1 - 0.36358 \cos{(z_n)} + 0.48917 \cos{(2z_n)} - 0.13659 \cos{(3z_n)} + 0.01064 \cos{(4z_n)}
```
Comment (alvgaona) [inline torch/signal/windows/windows.py:742]: @lezcano this formula computes each window timestep, no need to add `sum`.

Comment (lezcano) [reply torch/signal/windows/windows.py:742]: You are completely right. My bad!

**Review by lezcano — APPROVED** [review]
In it goes!

**Pull request closed without merging**